service/controller/resource/cleanuprecordsets/delete.go

```diff
@@ -0,0 +1,115 @@
+package cleanuprecordsets
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/route53"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/aws-operator/service/controller/controllercontext"
+	"github.com/giantswarm/aws-operator/service/controller/key"
+)
+
+func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCluster(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	var hostedZones []*route53.HostedZone
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finding hosted zones for tenant cluster %#q", key.ClusterID(&cr)))
```

Comment:
We should have cluster ID injected in the logs one way or another. I wouldn't print it in log messages.

Reply:
What is wrong with this way of logging? How can I do this in a different way?

Reply:
Just skip ` for tenant cluster %#q` part of the log.

Reply:
But in which way `finding hosted zones` message is informative then?

Reply:
Cluster ID is part of the reconciled object name which is already logged.